### ISSUE TITLE
Cannot package script with 'pydub'

### ISSUE THREAD

**Issue by need4swede** [pyinstaller/pyinstaller#7459]
System:

```
Macbook Air [M1, arm64]
macOS 13.1
Python 3.9.12
```

Code:

```
from pydub import AudioSegment

input_file = '/Users/${USER}/audio.mp3'
output_file = '/Users/${USER}/audio.wav'

sound = AudioSegment.from_mp3(input_file)
sound.export(output_file, format="wav")
```

This works if I run it through my IDE
This also works if I package it with PyInstaller `pyinstaller --onedir --windowed main.py` and run it from the executable. 
But it crashes if I run it through the .APP

I've tried including `--hidden-import pubdub` (and also 'pyaudioop') but to no avail

I suspect the error is caused due to some import error.
In my warning file, I see this: `missing module named pyaudioop - imported by pydub.audio_segment (top-level)`

I've tried myself to play around with the imports in pudub's files, but haven't been able to crack it.
Any help would be greatly appreciated! Searching the web shows I'm not alone with this issue, but I have yet to find a working solution.

Thank you!

**Comment by bwoodsend** [comment]
And what error message do you get if you launch your application from the terminal?

**Comment by bwoodsend** [comment]
My bad, I see you already said that it works when you do that.

**Comment by need4swede** [comment]
No worries! 

I came across this page on [Stack Overflow ](https://stackoverflow.com/questions/74448497/subprocess-problem-with-pydub-python-3-63-v-python-3-10#answer-74489986)that talked about pydub's use of subprocess

I made the changes suggested there, but still didn't resolve the issue.

EDIT: And thank you for your help!

**Comment by bwoodsend** [comment]
Invalid stream handles are indeed a common _windowed mode only_ issue but it's not this issue. Using poor man's _try/except/log error to file_:

```python
import traceback

try:
    from pydub import AudioSegment

    input_file = '/Users/Shared/08 Encased in Ice.mp3'
    output_file = '/Users/home/audio.wav'

    sound = AudioSegment.from_mp3(input_file)
    sound.export(output_file, format="wav")
except:
    open("/Users/home/error.log", "w").write(traceback.format_exc())
```

I get:
```python
Traceback (most recent call last):
  File "test.py", line 9, in <module>
  File "pydub/audio_segment.py", line 796, in from_mp3
  File "pydub/audio_segment.py", line 728, in from_file
  File "pydub/utils.py", line 274, in mediainfo_json
  File "subprocess.py", line 966, in __init__
  File "subprocess.py", line 1842, in _execute_child
FileNotFoundError: [Errno 2] No such file or directory: 'ffprobe'
```

This is expected because `ffprobe` is installed via homebrew into `/opt/homebrew/bin/ffprobe` and `/opt/homebrew/bin` is not in `PATH` for desktop applications - rather it's added to `PATH` in terminal sessions by your `.zshrc` (or in my case `config.fish` because I hate zsh).

This is a bit of design constraint of macOS - `.app` applications are intended to be self contained and not depend on anything from homebrew. Let me see if some `--add-binary` skulduggery can _vendor_ `ffprobe` into the executable...

**Comment by need4swede** [comment]
You're dead on!
Got the same error. I figured it was something not importing correctly - in hindsight, I probably should have started where you did, but I'd still be stuck at this point :P

Thanks a bunch - I really appreciate your help on this

**Comment by bwoodsend** [comment]
Vendoring `ffprobe` looks like a no go. If you run `otool -L /opt/homebrew/bin/ffprobe` you'll see just how many homebrew-provided libraries it depends on - there's no way to package them all and still stay sane by the end of it.

Honestly, I think that the closest that `pydub` is ever going to get to actually being usable by a `.app` is if you tell your users to install homebrew, install ffprobe using homebrew and then put something like `os.environ["PATH"] += os.pathsep + "/opt/homebrew/bin"` at the top of your code. Probably time to see if you can find another library which statically links against ffmpeg instead - [pyffmpeg](https://github.com/deuteronomy-works/pyffmpeg) looks like a good candidate.

**Comment by need4swede** [comment]
>Probably time to see if you can find another library which statically links against ffmpeg instead - [pyffmpeg](https://github.com/deuteronomy-works/pyffmpeg) looks like a good candidate.

Darn, that's a shame :/

Foolish of me to build an entire project around this library without testing the `.app`
Didn't think much of it, as it worked flawlessly in my testing. 

Is there no other way of 'duct taping' this project together, just for the sake of packaging it into an `.app`, apart from having users setup the environment?
I'll look into some alternatives in the mean time - thank you for sharing that info!

**Comment by rokm** [comment]
You could look around for statically built ffmpeg executables for macOS, and try bundling that with your .app? Not sure if those are available for arm64 yet, but if you have no other alternative, you could probably also get away with x86_64 version since it is ran as a subprocess.

**Comment by need4swede** [comment]
@bwoodsend @rokm 

Thank you both for chipping in and for helping me with this issue.
I've decided to go with @bwoodsend's suggestion and use something like `pyffmpeg` instead

I ran some sample tests with this library and it seems to package into an `.app` just fine.
It seems to take `ffmpeg` commands as an added bonus, which may actually help scale the project in the future.

Might not mean much, but I truly appreciate your time and your kindness.
Thank you very much.

Cheers!
